### PR TITLE
fix: ch media login behavior

### PR DESCRIPTION
**Pull request by sherwino**
### Description

This PR is a continuation of the work done on #532, where we make the necessary changes to mobile login.
Here we ensure we are saving the correct tokens to storage.

See screen capture of six different flows / scenarios with login plugin and profile selector.

https://drive.google.com/drive/folders/19Nn9aUPeJC2vMqziLL5w-YEVULbzSCK_?usp=sharing

### CODE REVIEW

File: plugins/zapp-ch-media-login/src/Components/Login/index.tsx [hunk 1]
```diff
@@ -101,12 +106,32 @@ const Login = (props) => {
...
+      }
+
+      const brandId = platformSelect({
+        ios: 1,
+        android: 105,
+        android_tv: 105,
+        tvos: 1
+      });
+
+      const params = `?deviceId:${contextData?.deviceMake}&brandId=${brandId}&name=${contextData?.deviceName}`;
```
Comment (ranmeirman) [inline plugins/zapp-ch-media-login/src/Components/Login/index.tsx:126]: Device Id should be the device Id provided after a successful login. Not the make. In addition, you have a redundant ` :`

**Review by ranmeirman — CHANGES_REQUESTED** [review]

File: plugins/zapp-ch-media-login/src/Components/Login/index.tsx (hunk 1 above)
Comment (sherwino) [reply plugins/zapp-ch-media-login/src/Components/Login/index.tsx]: Ok fixed this

File: plugins/zapp-ch-media-login/src/Components/Login/Utils/index.ts [hunk 2]
```diff
@@ -80,211 +83,260 @@ export const isAuthenticationRequired = (payload: ZappEntry) => {
       requires_authentication: !!requires_authentication,
     },
   });
+
   return !!requires_authentication;
 };
 
 export function isTokenExpired(expiresIn: number): boolean {
+  // TODO: maybe make a request, there are other ways token can be expired
   return expiresIn < moment().unix();
```
Comment (ranmeirman) [inline plugins/zapp-ch-media-login/src/Components/Login/Utils/index.ts:92]: Can we do expiresIn - 60 seconds? (if it is not done already)

File: plugins/zapp-ch-media-login/src/Components/Login/index.tsx [hunk 3]
```diff
@@ -260,7 +284,7 @@ const Login = (props) => {
                   backgroundColor: "transparent",
                 }}
                 source={{
-                  uri: loginUrl,
+                  uri: loginUrl + loginParams,
```
Comment (ranmeirman) [inline plugins/zapp-ch-media-login/src/Components/Login/index.tsx:287]: Can we store it on local var and log the url?

**Review by ranmeirman — CHANGES_REQUESTED** [review]
2 small changes

**Review by ranmeirman — APPROVED** [review]

**Pull request merged**